### PR TITLE
refactor: drop flat output config model

### PR DESCRIPTION
**Pull request by strawgate**
## Summary
- remove the flat `OutputConfig` / `OutputConfigEntry` compatibility layer
- store `PipelineConfig.outputs` as `Vec<OutputConfigV2>` and validate/build sinks directly from typed variants
- update generated `blast` config rendering and tests to construct typed output variants directly
- update config tests to assert typed-schema parse-time errors for invalid output fields/tags

## Tests
- `cargo fmt --check`
- `cargo check -p logfwd-config -p logfwd-runtime`
- `cargo test -p logfwd-config`
- `cargo test -p logfwd-runtime generated_cli --lib`
- `cargo test -p logfwd --bin ff blast -- --nocapture`
- `cargo clippy -p logfwd-config -p logfwd-runtime -p logfwd -- -D warnings`

> [!NOTE]
> ### Remove flat `OutputConfig` compatibility layer in favor of typed `OutputConfigV2`
> - Removes the flat `OutputConfig` struct, `OutputConfigEntry` wrapper, and all conversion/compatibility code across `logfwd-config`, `logfwd-runtime`, and `logfwd`.
> - `OutputConfigV2` gains `endpoint()`, `auth()`, `name()`, and `output_type()` accessor methods to replace fields previously exposed via the flat model.
> - `OutputType` gains `is_endpoint_required()` and `is_auth_supported()` helper methods used in validation and CLI resolution.
> - `PipelineConfig.outputs` and `RawConfig.output` now hold `OutputConfigV2` directly; validation and sink construction pass typed configs throughout.
> - Behavioral Change: serde error messages change (e.g., field paths like `output.type` or `pipelines.app.outputs[0]`); unknown fields are now rejected at deserialization time via `deny_unknown_fields` rather than via explicit compatibility checks.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized e895656.</sup>
> 
>

### CODE REVIEW

File: crates/logfwd-runtime/src/generated_cli.rs [hunk 1]
```diff
@@ -162,6 +206,40 @@ pub fn render_blast_yaml(
...
+        OutputConfigV2::Loki(config) => config.endpoint.as_deref(),
+        OutputConfigV2::Tcp(config) => config.endpoint.as_deref(),
+        OutputConfigV2::Udp(config) => config.endpoint.as_deref(),
+        OutputConfigV2::ArrowIpc(config) => config.endpoint.as_deref(),
+        OutputConfigV2::Stdout(_)
+        | OutputConfigV2::File(_)
+        | OutputConfigV2::Parquet(_)
+        | OutputConfigV2::Null(_) => None,
+        _ => None,
+    }
```
[the remark this answers is not in the document]
Comment (strawgate) [reply crates/logfwd-runtime/src/generated_cli.rs]: Addressed by moving `endpoint()` and `auth()` onto `OutputConfigV2` and replacing the duplicated generated CLI/test helpers with those methods.

File: crates/logfwd-config/src/types.rs [hunk 2]
```diff
@@ -1600,9 +1228,9 @@ pub struct PipelineConfig {
     pub inputs: Vec<InputConfig>,
     #[serde(default, deserialize_with = "deserialize_option_strict_string")]
     pub transform: Option<String>,
-    /// Typed output configurations accepted from V2 or legacy output YAML.
+    /// Typed output configurations.
     #[serde(default, deserialize_with = "deserialize_one_or_many")]
-    pub outputs: Vec<OutputConfigEntry>,
+    pub outputs: Vec<OutputConfigV2>,
```
[the remark this answers is not in the document]
Comment (strawgate) [reply crates/logfwd-config/src/types.rs]: Addressed by expanding the `PipelineConfig.outputs` field docs to describe `deserialize_one_or_many`, `OutputConfigV2` typed parsing, strict variant-specific rejection, and adding a compiling doctest for single and list output forms.

File: crates/logfwd-config/src/types.rs [hunk 3]
```diff
@@ -158,6 +158,30 @@ pub enum OutputType {
...
+    }
+
+    /// Return whether this output type supports HTTP-style authentication.
+    pub fn supports_auth(&self) -> bool {
+        matches!(
+            self,
+            Self::Otlp | Self::Http | Self::Elasticsearch | Self::Loki | Self::ArrowIpc
+        )
+    }
+}
```
[the remark this answers is not in the document]
Comment (strawgate) [reply crates/logfwd-config/src/types.rs:197]: Addressed by renaming the public boolean helpers to `is_endpoint_required()` / `is_auth_supported()` and adding compiling doctests for both.

File: crates/logfwd-config/src/types.rs [hunk 4]
```diff
@@ -1041,6 +907,40 @@ impl OutputConfigV2 {
...
+            OutputConfigV2::Loki(config) => config.auth.as_ref(),
+            OutputConfigV2::ArrowIpc(config) => config.auth.as_ref(),
+            OutputConfigV2::Stdout(_)
+            | OutputConfigV2::File(_)
+            | OutputConfigV2::Parquet(_)
+            | OutputConfigV2::Null(_)
+            | OutputConfigV2::Tcp(_)
+            | OutputConfigV2::Udp(_) => None,
+        }
+    }
```
[the remark this answers is not in the document]
Comment (strawgate) [reply crates/logfwd-config/src/types.rs:988]: Addressed by adding compiling doctests for `OutputConfigV2::endpoint()` and `OutputConfigV2::auth()` covering both `Some(...)` and `None` cases.

File: crates/logfwd-runtime/src/generated_cli.rs [hunk 5]
```diff
@@ -62,59 +66,83 @@ pub fn resolve_blast_output_config(
...
+            endpoint,
+            ..Default::default()
+        }),
+        OutputType::ArrowIpc => OutputConfigV2::ArrowIpc(ArrowIpcOutputConfig {
+            endpoint,
+            auth,
+            ..Default::default()
+        }),
+        _ => return Err(format!("unsupported blast destination '{output_type}'")),
+    })
```
[the remark this answers is not in the document]
Comment (strawgate) [reply crates/logfwd-runtime/src/generated_cli.rs:140]: Addressed by rejecting `http` and `parquet` in `resolve_blast_output_config` with the unsupported-destination error before building generated YAML/config.

File: crates/logfwd-runtime/src/generated_cli.rs [hunk 6]
```diff
@@ -130,12 +158,12 @@ pub fn render_blast_yaml(
     let _ = writeln!(yaml, "          batch_size: {batch_lines}");
     yaml.push_str("          events_per_sec: 0\n");
     yaml.push_str("    outputs:\n");
-    let _ = writeln!(yaml, "      - type: {}", output_cfg.output_type);
+    let _ = writeln!(yaml, "      - type: {}", output_cfg.output_type());
```
[the remark this answers is not in the document]
Comment (strawgate) [reply crates/logfwd-runtime/src/generated_cli.rs]: Addressed by quoting the rendered output type with `yaml_quote(...)` and adding a generated-CLI test proving `OutputType::Null` renders as `type: "null"` and parses successfully.

File: crates/logfwd-runtime/src/generated_cli.rs [hunk 7]
```diff
@@ -62,59 +66,81 @@ pub fn resolve_blast_output_config(
...
-            | OutputType::Tcp
-            | OutputType::Udp
-    ) {
-        let endpoint =
-            endpoint.ok_or_else(|| "blast requires --endpoint for this destination".to_owned())?;
-        output_cfg.endpoint = Some(endpoint.to_owned());
+) -> Result<OutputConfigV2, String> {
+    if matches!(output_type, OutputType::Http | OutputType::Parquet) {
+        return Err(format!("unsupported blast destination '{output_type}'"));
     }
```
[the remark this answers is not in the document]
Comment (strawgate) [reply crates/logfwd-runtime/src/generated_cli.rs]: Addressed in `e8956565`: `blast` now rejects `file` upfront alongside `http`/`parquet`, removes the default `FileOutputConfig` construction path, and the generated CLI rejection test covers `OutputType::File`.

File: crates/logfwd-core/src/structural_iter.rs [hunk 8]
```diff
@@ -449,6 +449,18 @@ mod verification {
...
+                | p.real_quotes
+                | p.comma
+                | p.colon
+                | p.open_brace
+                | p.close_brace
+                | p.open_bracket
+                | p.close_bracket)
+                & mask
+                != 0,
+        );
```
[the remark this answers is not in the document]
Comment (strawgate) [reply crates/logfwd-core/src/structural_iter.rs]: Addressed in `e8956565`: updated the proof documentation, kept the structural-bit precondition explicit, and added two `kani::cover!` guards. Verified with the targeted `verify_classify_bit_correct` harness; both covers are satisfied.

File: crates/logfwd-config/src/types.rs [hunk 9]
```diff
@@ -1298,24 +1106,12 @@ pub struct StdoutOutputConfig {
...
-
 #[derive(Debug, Clone, PartialEq, Eq, Deserialize, Default)]
 #[serde(deny_unknown_fields)]
 /// File output configuration for the typed V2 schema.
 ///
-/// File compression is intentionally not part of the typed schema. Legacy flat
-/// configs may still carry `compression`; compatibility code preserves that
-/// field in the flat view so it can reject the unsupported option explicitly.
+/// File compression is intentionally not part of the typed schema; unknown
+/// fields are rejected during deserialization.
```
[the remark this answers is not in the document]
Comment (strawgate) [reply crates/logfwd-config/src/types.rs:1114]: Addressed in `e8956565`: added a compiling `FileOutputConfig` doctest that parses a file output through `Config::load_str` and asserts `OutputConfigV2::File`. Verified with `cargo test -p logfwd-config --doc`.
